### DISCUSSION TITLE
Suggested workarounds for discovering Terragrunt dependency changes during plans?

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#552] [Help] [answered]
If I have the following configuration, where my Aurora cluster is dependent on my VPC, how can I let the Aurora cluster know during the `plan` phase that it should also be recreated? 

```hcl
# vpc/terragrunt.hcl 

terraform {
  source = "git@github.com:iangrunt/terraform-fake-modules.git//modules/aws/vpc?ref=v1.0.0"
}

inputs = {
  # Old value at the time when we applied:
  #  cidr_block = "10.111.0.0/16"

  # New value that will recreate my VPC:
  cidr_block = "10.222.0.0/16"
}

```

```hcl
# aurora-postgres/terragrunt.hcl 

terraform {
  source = "git@github.com:iangrunt/terraform-fake-modules.git//modules/aws/aurora?ref=v1.0.0"
}

dependency "vpc" {
  config_path = "../vpc"
}

inputs = {
  vpc_id = dependency.vpc.outputs.id
}
```

When I run a `terragrunt run-all plan` I see my VPC being recreated correctly as the `cidr_block` has changed to force that, but I need to let my Aurora cluster know it will also be recreated:

<img width="1025" alt="Screen Shot 2022-09-02 at 12 46 34 PM" src="https://user-images.githubusercontent.com/101607944/188202083-713409b3-645e-4872-9c01-b9743064c419.png">

---

<ins datetime="2022-09-02T16:55:32Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109200">Tracked in ticket #109200</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
This is unfortunately a known limitation of `run-all plan` where there is no clear workaround we have implemented yet. This specific issue and suggested alternative workflows are documented in the associated warning for `run-all plan` in [the documentation](https://terragrunt.gruntwork.io/docs/reference/cli-options/#run-all).
